Source/WebCore/crypto/mac/CryptoKeyOKPCocoa.cpp

```diff
@@ -39,12 +39,12 @@ namespace WebCore {
 
 bool CryptoKeyOKP::isPlatformSupportedCurve(NamedCurve namedCurve)
 {
-    return namedCurve == NamedCurve::Ed25519;
+    return namedCurve == NamedCurve::Ed25519 || namedCurve == NamedCurve::X25519;
 }
 
 std::optional<CryptoKeyPair> CryptoKeyOKP::platformGeneratePair(CryptoAlgorithmIdentifier identifier, NamedCurve namedCurve, bool extractable, CryptoKeyUsageBitmap usages)
 {
-    if (namedCurve != NamedCurve::Ed25519)
+    if (namedCurve != NamedCurve::Ed25519 && namedCurve != NamedCurve::X25519)
```

Comment:
These checks could be replaced by directly calling `isPlatformSupportedCurve()`. It would simplify things, especially if in the future there's also the need to similarly test for X448 and Ed448.

Reply:
It seems similar changes should be done in the GCrypt based implementation, using by the GTK+  port, so I believe this suggestion could be addressed as part of a general refactoring patch later.